vowpalwabbit/core/src/reductions/details/automl_impl.h

```diff
@@ -165,7 +171,8 @@ struct interaction_config_manager : config_manager
   // Stores estimators of live configs, size will never exceed max_live_configs. Each pair will be of the form
   // <challenger_estimator, champ_estimator> for the horizon of a given challenger. Thus each challenger has one
   // horizon and the champ has one horizon for each challenger
-  std::vector<std::pair<aml_estimator, estimator_config>> estimators;
+  std::vector<challenger_champ_pair> estimators;
+  std::vector<std::pair<aml_estimator<estimator_impl>, estimator_impl>> estimators_2;
```

Comment:
@bassmang this is what i mentioned earlier, adding seperate var for now before committing to it